google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/stream.rb

```diff
@@ -295,6 +300,19 @@ def background_run
 
           # rubocop:enable all
 
+          def create_receipt_modack_for_eos received_messages
+            received_messages.each do |rec_msg_grpc|
+              callback = proc do |result|
+                if result.succeeded?
+                  synchronize { @inventory.add rec_msg_grpc }
+                  rec_msg = ReceivedMessage.from_grpc rec_msg_grpc, self
+                  register_callback rec_msg
+                end
+              end
+              @subscriber.buffer.modify_ack_deadline @subscriber.deadline, [rec_msg_grpc.ack_id], callback
```

Comment:
I can't tell where the temp failures are retried. Can you help me follow, maybe a link to the code that will eventually trigger a retry for this ack id?

Reply:
handling of errors happen [here](https://github.com/googleapis/google-cloud-ruby/blob/9410d03deeaff08b605302b8937248a2b82d7995/google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/timed_unary_buffer.rb#L149)